homematic-devices/hmip-drdi3.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-new */
+
+const Accessory = require('./lib/accessory');
+
+function addService(type, name, channel) {
+    let service
+    this.node.debug("HMIP-DRDI3: "+ name +" "+channel);
+    switch (type) {
+        case 'ValveIrrigation':
+        // intentional fallthrough
+        case 'Valve':
+            // intentional fallthrough            
+        case 'Lightbulb':
+            service=this.addService(type,name);
+            service.get('On', channel + '.LEVEL', value => {
+                this.node.debug("HMIPW-DRD3 get : "+channel);
+                valueBrightness = value;
+                return value > 0;
+            })
+
+            service.set('On', (value, callback) => {
+                this.node.debug("HMIPW-DRD3 set : "+channel);
+                if (value) {
+                    setTimeout(() => {
+                        if (valueBrightness === 0) {
+                            value = 1;
+                        } else {
+                            value = valueBrightness / 100;
+                        }
+
+                        this.ccuSetValue(channel + '.LEVEL', value, callback);
+                    }, 100);
+                } else {
+                    this.ccuSetValue(channel + '.LEVEL', 0, callback);
+                }
+            })
+
+            service.get('Brightness', channel + '.LEVEL', value => {
+                valueBrightness = value * 100;
+                return value * 100;
+            })
+
+            service.set('Brightness', channel + '.LEVEL', value => {
+                valueBrightness = value;
+                return value / 100;
+            });
+        case 'Fan':
+        // intentional fallthrough
+        case 'Outlet':
+        // intentional fallthrough
+        default:
+            /*this.addService(type, name, type === 'Switch' ? '' : type)
+                .get('On', dp)
+                .set('On', dp);*/
+    }
+}
+
+class AccSingleService extends Accessory {
+    init(config, node) {
+        const {ccu} = node;       
+        node.debug(config.accChannel + ' ' + "Lightbulb" + ' ');
+        addService.call(this, "Lightbulb", config.accChannelName, config.accChannel);
+    }
+}
+
+class AccMultiService extends Accessory {
+    init(config, node) {
+        const {ccu} = node;
+        
+        for (let j = 0; j < 3; j++) {
+            for (let c = 0; c < 3; c++) {
+                const i = (j * 4) + (c + 2);
```

Comment:
Da der erste verwertbare Kanal der 5. ist, müsste hier eine +5 statt einer +2 verwendet werden?

Reply:
Durch die Virtual Channels sieht man auch die verdeckten Kanäle.
Der Dimmer hat auf 1,5,9 die Statuskanäle als Rückgabe. Deswegen ist der erste Kanal 2,6,10 für Ausgang 1,2,3
habe ein Bild meiner CCU3 gemacht dort erkennt man es.
![drd3](https://user-images.githubusercontent.com/44172078/210168083-7937af50-8305-43fd-a0f7-5ecd7e722c39.PNG)

Reply:
Du hast einen DRD3, ich einen DRDI3, sind die vielliecht unterschiedlich?
So sieht es bei mir aus:
![grafik](https://user-images.githubusercontent.com/10615289/210168311-06c11c47-df1c-47b1-b5c3-f7c633d3b931.png)

Ohne die Anpassung in der Datei `redmatic-homekit-homematic-devices.html` lief es bei mir jedenfalls definitiv nicht. Den Code in dieser Datei verstehe ich allerdings nicht zu 100%.

Reply:
Das Kann sein.
Dein DRDI3 hat tastereingänge. hast du die virtuellen Kanäle an?

Reply:
Also ich nutze zumindest bislang keine virtuellen Kanäle. Die Einstellung "Kanal aktiv / inaktiv" bei den Statusmitteilungs-Kanälen steht auf "Aktiv", falls du das meinst.

Reply:
meinte das hier,
es gibt bei Homematic immer 3 Kanäle pro Aktor Ausgang. es wäre interessant wie es dann aussieht wenn dort  die virtuellen Kanäle (erweiterter Verknüpfungsmodus an ist). 
Wenn du nur den Start Kanal anpassen musstest. dann passt das. Die Tastereingänge werden aktuell nicht ausgewertet

Reply:
Achso, das ist wie bei dir nicht angehakt.

Reply:
Also bei mir läuft jetzt alles wie es soll, an beiden Stellen muss für den DRDI3 aus der 2 eine 5 gemacht werden. Vielen Dank für die Vorarbeit!